neuro_san/internals/run_context/langchain/core/run_context_runnable.py

```diff
@@ -80,51 +69,20 @@ class NeuroSanRunnable(RunnablePassthrough):
 
     primary_llm: BaseLanguageModel
 
-    invocation_context: InvocationContext
-
     journal: Journal
 
-    interceptor: InterceptingJournal
-
-    origin: List[Dict[str, Any]]
-
     tool_caller: ToolCaller
 
     error_detector: ErrorDetector
 
     session_id: str
 
-    # Default logger
-    logger: Optional[Logger] = None
-
     # This guy needs to be a pydantic class and in order to have
     # a non-pydantic Journal as a member, we need to do this.
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
-    def __init__(
-        self,
-        **kwargs: Any,
-    ) -> None:
-        """
-        Constructor
-        """
-        super().__init__(afunc=self.do_it, **kwargs)
-
-    # pylint: disable=redefined-builtin
-    @override
-    async def ainvoke(
-        self,
-        input: Other,
-        config: RunnableConfig | None = None,
-        **kwargs: Any | None,
-    ) -> Other:
-
-        _: Other = await super().ainvoke(input, config, **kwargs)
-        outputs: Dict[str, Any] = self.get_intercepted_outputs()
-        return outputs
-
     # pylint: disable=redefined-builtin
-    async def do_it(self, inputs: Input) -> Output:
+    async def run_it(self, inputs: Input) -> Output:
```

Comment:
Rename the entrypoint method and combine with what main_invoke() was doing before.